modules/ztp-configuring-ptp-fast-events.adoc

```diff
@@ -0,0 +1,97 @@
+|====
+
+.Prerequisites
+
+* Create a git repo where you manage your custom argoCD site configuration data.
+* Extract the contents of the `/home/ztp` folder from the `quay.io/redhat_emp1/ztp-site-generator:latest` reference architecture container image, and review the changes.
+
+.Procedure
+
+. Add the following YAML into `.spec.sourceFiles` in the `common-ranGen.yaml` file to configure the AMQP Operator:
++
+[source,yaml]
+----
+#AMQ interconnect operator for fast events
+- fileName: AmqSubscriptionNS.yaml
+  policyName: "amq-sub-policy"
+- fileName: AmqSubscriptionOperGroup.yaml
+  policyName: "amq-sub-policy"
+- fileName: AmqSubscription.yaml
+  policyName: "amq-sub-policy"
+----
+
+. Apply the following `PolicyGenTemplate` changes to `group-du-3node-ranGen.yaml`, `group-du-sno-ranGen.yaml` or `group-du-standard-ranGen.yaml` files according to your requirements:
+
+.. In `.sourceFiles`, change the `config-policy` to reference the `PtpOperatorConfig` CR that configures the AMQ transport host:
++
+[source,yaml]
+----
+- fileName: PtpOperatorConfigForEvent.yaml
+  policyName: "config-policy"
+----
+
+.. Configure the `linuxptp` and `phc2sys` for the PTP clock type and interface. For example, add the following stanza into `.sourceFiles`:
++
+[source,yaml]
+----
+- fileName: PtpConfigSlave.yaml                   <1>
+  policyName: "ptp-config-policy"
+  metadata:
+    name: "du-ptp-slave"
+  spec:
+    profile:
+    - name: "slave"
+      interface: "ens5f1"                         <2>
+      ptp4lOpts: "-2 -s --summary_interval -4"    <3>
+      phc2sysOpts: "-a -r -m -n 24 -N 8 -R 16"
+    ptpClockThreshold:                            <4>
+      holdOverTimeout: 30 #secs
+      maxOffsetThreshold: 100  #nano secs
+      minOffsetThreshold: -100 #nano secs
+----
+<1> Can be one `PtpConfigMaster.yaml`, `PtpConfigSlave.yaml`, or `PtpConfigSlaveCvl.yaml` depending on your requirements. `PtpConfigSlaveCvl.yaml` configes `linuxptp` services for an Intel E810 Columbiaville NIC.
+<2> Device specific interface name.
+<3> You must append the `--summary_interval -4` value to `ptp4lOpts` in `.spec.sourceFiles.spec.profile` to enable PTP fast events. This value must also match the value set for `logSyncInterval` in the corresponding `PtpConfig` CR.
+<4> Optional.
+
+. Merge any other required changes and files with your custom site repo.
+
+. Push the changes to your site configuration repo to deploy PTP fast events to new sites using argoCD.
+
+. To deploy PTP events to an an existing site, use the Topology-Aware Lifecycle Operator to deploy the changes.
+
+[NOTE]
+====
+When the changes are pushed to a site configuration repo, argoCD applies the CRs sequentially according to the `ran.openshift.io/ztp-deploy-wave` annotation. When two or more policies have the same `ran.openshift.io/ztp-deploy-wave` annotation, an alphanumeric sort order is used.
```

Comment:
This note is generally true across any CRs added to the PolicyGenTemplates. Is there a way to reference back to the general GitOps ZTP documentation to avoid duplication?

Reply:
I moved the note and added a link to the relevant section in the assembly.